### PR TITLE
Update gRPC troubleshooting with HttpClient.Timeout

### LINKED ISSUE #33148: gRPC Troubleshoot:  Add section: gRPC call timeout from HttpClient.Timeout
### Description

Creating this issue to track PR already created and in review.

Also:

- Move code to the working app sample for this topic.
- Add the same new section to all versions of the topic.
- Some edits for readability.

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/grpc/troubleshoot?view=aspnetcore-8.0

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/grpc/troubleshoot.md

### Document ID

84ddcb67-03be-655c-470a-f43a5152606f

### Article author

@JamesNK

### PR DESCRIPTION
**Pull request by JamesNK**
From offline conversation.

Fixes #33148

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/grpc/troubleshoot.md](https://github.com/dotnet/AspNetCore.Docs/blob/dafd1323d8d2c426964d3a53e9276590a2ac4a02/aspnetcore/grpc/troubleshoot.md) | [Troubleshoot gRPC on .NET](https://review.learn.microsoft.com/en-us/aspnet/core/grpc/troubleshoot?branch=pr-en-us-33141) |

### CODE REVIEW

**Comment by JamesNK** [comment]
This content applies to all .NET versions and should be added for all monikers.

File: aspnetcore/grpc/troubleshoot.md [hunk 1]
```diff
@@ -207,6 +207,37 @@ dotnet build
 
 For more information about using `Grpc.Tools` with unsupported architectures, see the [gRPC build integration documentation](https://github.com/grpc/grpc/blob/master/src/csharp/BUILD-INTEGRATION.md#using-grpctools-with-unsupported-architectures).
 
+## gRPC call timeout from `HttpClient.Timeout`
+
+<xref:System.Net.Http.HttpClient> is configured with a 100 second timeout by default. If a `GrpcChannel` is configured to use a `HttpClient` then long running gRPC streaming calls are canceled if they don't completed don't complete within the timeout limit.
```
Comment (wadepickett) [inline aspnetcore/grpc/troubleshoot.md:212]:
```suggestion
<xref:System.Net.Http.HttpClient> is configured with a 100 second timeout by default. If a `GrpcChannel` is configured to use a `HttpClient` then long running gRPC streaming calls are canceled if they don't complete within the timeout limit.
```

File: aspnetcore/grpc/troubleshoot.md [hunk 2]
```diff
@@ -207,6 +207,37 @@ dotnet build
 
 For more information about using `Grpc.Tools` with unsupported architectures, see the [gRPC build integration documentation](https://github.com/grpc/grpc/blob/master/src/csharp/BUILD-INTEGRATION.md#using-grpctools-with-unsupported-architectures).
 
+## gRPC call timeout from `HttpClient.Timeout`
+
+<xref:System.Net.Http.HttpClient> is configured with a 100 second timeout by default. If a `GrpcChannel` is configured to use a `HttpClient` then long running gRPC streaming calls are canceled if they don't complete within the timeout limit.
```
Comment (wadepickett) [inline aspnetcore/grpc/troubleshoot.md:212]:
```suggestion
<xref:System.Net.Http.HttpClient> is configured with a 100 second timeout by default. If a `GrpcChannel` is configured to use an `HttpClient`, long-running gRPC streaming calls are canceled if they don’t complete within the timeout limit.
```

File: aspnetcore/grpc/troubleshoot.md [hunk 3]
```diff
@@ -207,6 +207,37 @@ dotnet build
...
+
+> System.OperationCanceledException: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.
```
Comment (wadepickett) [inline aspnetcore/grpc/troubleshoot.md:214]:
```suggestion
```output
> System.OperationCanceledException: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.
```
```

**Review by wadepickett — APPROVED** [review]
Some minor suggestions on the subject intro.

I also put System.OperationCanceledException in an `Output` code block.

If accepted, the updated section would also drop into ~/grpc/troubleshoot/includes/troubleshoot3-7.md)

**Comment by wadepickett** [comment]
moving code examples to the client sample for this topic, testing, and will add to this PR shortly....

**Comment by wadepickett** [comment]
@JamesNK, if the changes I made per your request look good, it can be merged.

File: aspnetcore/grpc/troubleshoot.md [hunk 4]
```diff
@@ -207,6 +207,22 @@ dotnet build
 
 For more information about using `Grpc.Tools` with unsupported architectures, see the [gRPC build integration documentation](https://github.com/grpc/grpc/blob/master/src/csharp/BUILD-INTEGRATION.md#using-grpctools-with-unsupported-architectures).
 
+## gRPC call timeout from `HttpClient.Timeout`
+
+<xref:System.Net.Http.HttpClient> is configured with a 100 second timeout by default. If a `GrpcChannel` is configured to use an `HttpClient`, long-running gRPC streaming calls are canceled if they don’t complete within the timeout limit.
+
+```output
+> System.OperationCanceledException: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.
```
Comment (JamesNK) [inline aspnetcore/grpc/troubleshoot.md:215]:
```suggestion
System.OperationCanceledException: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.
```

File: aspnetcore/grpc/troubleshoot/includes/troubleshoot3-7.md [hunk 5]
```diff
@@ -324,6 +324,22 @@ dotnet build
 
 For more information about using `Grpc.Tools` with unsupported architectures, see the [gRPC build integration documentation](https://github.com/grpc/grpc/blob/master/src/csharp/BUILD-INTEGRATION.md#using-grpctools-with-unsupported-architectures).
 
+## gRPC call timeout from `HttpClient.Timeout`
+
+<xref:System.Net.Http.HttpClient> is configured with a 100 second timeout by default. If a `GrpcChannel` is configured to use an `HttpClient`, long-running gRPC streaming calls are canceled if they don’t complete within the timeout limit.
+
+```output
+> System.OperationCanceledException: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.
```
Comment (JamesNK) [inline aspnetcore/grpc/troubleshoot/includes/troubleshoot3-7.md:332]:
```suggestion
System.OperationCanceledException: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.
```

File: aspnetcore/grpc/troubleshoot/includes/troubleshoot3-7.md [hunk 6]
```diff
@@ -557,6 +573,22 @@ The preceding code:
 
 Alternatively, a client factory can be configured with `SubdirectoryHandler` by using <xref:Microsoft.Extensions.DependencyInjection.HttpClientBuilderExtensions.AddHttpMessageHandler%2A>.
 
+## gRPC call timeout from `HttpClient.Timeout`
+
+<xref:System.Net.Http.HttpClient> is configured with a 100 second timeout by default. If a `GrpcChannel` is configured to use an `HttpClient`, long-running gRPC streaming calls are canceled if they don’t complete within the timeout limit.
+
+```output
+> System.OperationCanceledException: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.
```
Comment (JamesNK) [inline aspnetcore/grpc/troubleshoot/includes/troubleshoot3-7.md:581]:
```suggestion
System.OperationCanceledException: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.
```

File: aspnetcore/grpc/troubleshoot/includes/troubleshoot3-7.md [hunk 7]
```diff
@@ -790,4 +822,20 @@ The preceding code:
...
```
Comment (JamesNK) [inline aspnetcore/grpc/troubleshoot/includes/troubleshoot3-7.md:830]:
```suggestion
System.OperationCanceledException: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing.
```

**Pull request merged**